### PR TITLE
Adr/openapi

### PR DESCRIPTION
**Pull request by johnnolan**

### CODE REVIEW

File: src/adr/articles/0006-openapi.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Context
+
+We need a way to communicate our APIs intent to consumers. We should display all options available as well as required data and validation associated with integrating with our APIs.
+
+Any changes to the schema of our API should be testable via this specification.
+
+## Technical
+
+### Interoperability - How does this enable the exchange of information
```
Comment (mattmachell) [inline src/adr/articles/0006-openapi.md:22]: Worth referencing the https://www.gov.uk/government/collections/api-design-guidance

File: src/adr/articles/0006-openapi.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+OpenAPI is a very common format used across a wide range of services.
+
+There is great support for it through services and tool that we can take advantage of.
```
Comment (mattmachell) [inline src/adr/articles/0006-openapi.md:26]: Give some examples of the tools and what they make easy.

File: src/adr/articles/0006-openapi.md (hunk 1 above)
Comment (johnnolan) [reply src/adr/articles/0006-openapi.md:22]: Added and linked that reference

File: src/adr/articles/0006-openapi.md (hunk 2 above)
Comment (johnnolan) [reply src/adr/articles/0006-openapi.md]: Linked to openapi tools and an example list. Whats your feeling on linking to a non govuk site?

File: src/adr/articles/0006-openapi.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+**Overall**: 8/10
+Many of our teams have implemented OpenAPI specs in their services. Those that haven't do have knowledge of RESTful APIs and JSON or YAML to be able to work with it.
+
+### Support/Open Source - Is it well supported
+
+OpenAPI is a open source collaboration project.
+
+### Scalability
+
+N/A
```
Comment (mattmachell) [inline src/adr/articles/0006-openapi.md:51]: Room here to talk about how dealing with many services with many api definitions may impact scaling.

File: src/adr/articles/0006-openapi.md (hunk 2 above)
Comment (mattmachell) [reply src/adr/articles/0006-openapi.md]: To be encouraged.

File: src/adr/articles/0006-openapi.md (hunk 3 above)
Comment (johnnolan) [reply src/adr/articles/0006-openapi.md]: Added content in for this section

File: src/adr/articles/0006-openapi.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+- Automatic Documentation Generators
+- Mock Servers
+- Testing
+- Data Validators
+- CI Automation checks
+
+You can use JSON or YAML format for the structure and both are well known standards.
+
+GOV.UK have excellent [guidance on how you should document your APIs](https://www.gov.uk/guidance/how-to-document-apis) in there [API Design Guidance](https://www.gov.uk/government/collections/api-design-guidance) page.
```
Comment (hawx) [inline src/adr/articles/0006-openapi.md:38]: there=>their

File: src/adr/articles/0006-openapi.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Any changes to the schema of our API should be testable via this specification.
+
+## Technical
+
+### Interoperability - How does this enable the exchange of information
+
+OpenAPI is a common format used across a wide range of services.
+
+There is great support for OpenAPI through third party services and tools that we can take advantage of. A great example of tools available can be found on [OpenAPI.Tools](https://openapi.tools/).
```
Comment (mattmachell) [inline src/adr/articles/0006-openapi.md:26]: Worth adding a note on language support in our chosen languages.

**Review by mattmachell — DISMISSED** [review]

**Review by mattmachell — APPROVED** [review]

**Pull request merged**